// @atlaskit/modal-dialog/dist/cjs/internal/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.warningColor = exports.dangerColor = exports.focusOutlineColor = exports.keylineColor = exports.keylineHeight = exports.titleMargin = exports.actionMargin = exports.modalPadding = exports.verticalOffset = exports.gutter = exports.WIDTH_ENUM = void 0;

var _colors = require("@atlaskit/theme/colors");

var _constants = require("@atlaskit/theme/constants");

var WIDTH_ENUM = {
  values: ['small', 'medium', 'large', 'x-large'],
  widths: {
    small: 400,
    medium: 600,
    large: 800,
    'x-large': 968
  },
  defaultValue: 'medium'
};
exports.WIDTH_ENUM = WIDTH_ENUM;
var gutter = 60;
exports.gutter = gutter;
var verticalOffset = (0, _constants.gridSize)() * 2;
exports.verticalOffset = verticalOffset;
var modalPadding = (0, _constants.gridSize)() * 3;
exports.modalPadding = modalPadding;
var actionMargin = (0, _constants.gridSize)();
exports.actionMargin = actionMargin;
var titleMargin = (0, _constants.gridSize)();
exports.titleMargin = titleMargin;
var keylineHeight = 2;
exports.keylineHeight = keylineHeight;
var keylineColor = _colors.N30;
exports.keylineColor = keylineColor;
var focusOutlineColor = _colors.N800;
exports.focusOutlineColor = focusOutlineColor;
var dangerColor = _colors.R400;
exports.dangerColor = dangerColor;
var warningColor = _colors.Y400;
exports.warningColor = warningColor;
